File: python/leetcode/0007-reverse-integer/solution01.py
MAX_INTERGER32 = 2 ** 31


class Solution:
    def reverse(self, x: int) -> int:
        multply = 1
        if x < 0:
            multply = -1

        x *= multply
        reversed = 0
        while x > 0:
            reversed = reversed * 10 + x % 10
            if reversed > MAX_INTERGER32:
                return 0
            x = x // 10

        return reversed * multply

File: typescript/leetcode/7-reverse-integer/solution01.ts
/**
 * @param {number} x
 * @return {number}
 */
export function reverse(x: number): number {
  const multiply = x < 0 ? -1 : 1
  x *= multiply

  let reversed = 0
  while (x > 0) {
    reversed = reversed * 10 + (x % 10)
    if (reversed > MAX_INTEGER32) {
      return 0
    }
    x = Math.floor(x / 10)
  }

  return reversed * multiply
}

const MAX_INTEGER32 = 0x7fffffff // Math.pow(2, 31)
